portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -28,29 +39,81 @@
-  * @param  response The <code>HttpServletResponse</code> for the POST request.
-  * @return None. A Entity of the Comment kind is created and upserted to the Datastore.
-  */
+   * This Method handles POST requests corresponding to a new comment and creates a new Entity for
+   * that comment in the Google Cloud Datastore.
+   *
+   * <p>This POST request originates from the 'new comment' form in server-dev.html and is initially
+   * sent to Blobstore for file processing. Once the Blobstore forward the request to this servlet,
+   * the name of file submitted in the form can be used to get the image URL to be stored in
+   * Blobstore. The POST request also results in a re-direct back to the original server-dev page.
+   * TODO(Issue #15): Do verfification on a new comment before adding it to the comments list.
+   *
+   * @param request The <code>HttpServletRequest</code> for the POST request.
+   * @param response The <code>HttpServletResponse</code> for the POST request.
+   * @return void. A Entity of the Comment kind is created and upserted to the Datastore.
+   */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String newComment = request.getParameter("comment");
     long timestamp = System.currentTimeMillis();
+    String imageUrl = getUploadedFileUrl(request, "image");
 
     Entity taskEntity = new Entity("Comment");
     taskEntity.setProperty("text", newComment);
     taskEntity.setProperty("timestamp", timestamp);
-
+    taskEntity.setProperty("imageUrl", timestamp);
     datastore.put(taskEntity);
 
     response.sendRedirect("/pages/server-dev.html");
   }
+
+  /**
+   * Returns a String corresponding to the URL that points to the uploaded file, or null if the user
+   * didn't upload a file.
+   *
+   * @param request The <code>HttpServletRequest</code> for the POST request.
+   * @param formInputElementNameThe Name attribute of the image file input to the form.
+   * @return The URL that points to the uploaded file.
+   */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // Check the validity of the file here by making sure it's an image file
+    if (!"image".equals(blobInfo.getContentType().substring(0, 5))) {
```

Comment:
I am suggesting to find a enum class for this MIME file type to replace \"image\" here. but I am ok with this since Interestingly this getContentType() returns a String, which should return a ENUM.

Reply:
I did find [this](https://guava.dev/releases/19.0/api/docs/com/google/common/net/MediaType.html#ANY_IMAGE_TYPE) but I am not sure that it is what I am looking for. So since it returns a String instead of an enum and there doesn't seem to be a nice way to get the image content type enum I will keep it as is. Please let me know if you think there is a better solution and I can do some more research.

Reply:
Good finds! Yeah ideally we should use string from ENUM to match the type but your implementation lgtm. just want to show a better way in case you need to do something similar in the future.